### PR TITLE
Add daily build and build pipeline improvements

### PR DESCRIPTION
**Pull request by MadhurangaWije**
## Purpose
- Add daily build 
## Release Note
- Daily builds for all the OpenAPI generated connectors
- Publish daily build artifacts to GitHub Packages
- Identify the change whenever a change occurred in any file of the package
- Support for SNAPSHOT versioning
- Add Pull Request template

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -107,41 +119,55 @@ updatedBallerinaPackages = Utils.findUpdatedBallerinaPackages(openApiPackageDirP
...
+        //     exec {
+        //         commandLine 'sh', '-c', "rm -rf ${ballerinaDistributionPath}/bin/../dependencies/jdk-11.0.8+10-jre"
+        //     }
+
+        //     // Changing the build distribution to SL Alpha5
+        //     exec {
+        //         commandLine 'sh', '-c', "${ballerinaDistributionPath}/bin/bal dist use slalpha5"
+        //     }
+        // }
+    // }
```
Comment (LakshanSS) [inline build.gradle:138]: Any reason for having commented code?

File: build.gradle [hunk 2]
```diff
@@ -152,3 +178,70 @@ task releaseConnector {
...
     }
 }
+
+// Daily build configurations
+def packageOrg = "ballerinax"
+def platform = "java11"
+def artifactCacheParent = file("${project.projectDir}/build/cache_parent/")
+
+def packageName = "";
+def packageVersion = "";
```
Comment (LakshanSS) [inline build.gradle:188]: If these are only used within copyDistribution task, we can define these within that task.

File: build.gradle [hunk 3]
```diff
@@ -152,3 +178,70 @@ task releaseConnector {
...
+                into file("${artifactCacheParent}/cache/")
+            }
+            outputs.dir artifactCacheParent
+        }
+    }
+}
+
+task createArtifactZip(type: Zip) {
+    if (isPublishing){
+        println("Create Zip Artifact");
```
Comment (LakshanSS) [inline build.gradle:218]: Shall we change the message as 'Creating Zip Artifact' ?

File: build.gradle [hunk 4]
```diff
@@ -152,3 +178,70 @@ task releaseConnector {
...
+        from "${buildDir}/cache_parent"
+        archiveName 'distribution.zip'
+        destinationDir(file("${buildDir}/distribution"))
+    }    
+}
+
+publishing {
+    if (isPublishing){
+        println("Package published");
```
Comment (LakshanSS) [inline build.gradle:227]: Shall we change this as 'Publishing package'?

File: openapi/covid19/Dependencies.toml [hunk 5]
```diff
@@ -1,9 +1,11 @@
...
-version = "1.1.0-alpha8"
+version = "1.1.0-beta.2"
 
 [[dependency]]
 org = "ballerina"
 name = "url"
-version = "1.1.0-alpha8"
+version = "1.1.0-beta.2"
+
+
```
Comment (LakshanSS) [inline openapi/covid19/Dependencies.toml:11]: Only one new line should be added in the end of a file.

File: build.gradle (hunk 1 above)
Comment (MadhurangaWije) [reply build.gradle:138]: Yes, these commented code is related to code generation, and commented out at the moment as a temporary fix untill the issues in the OpenAPI build tool get fixes.

**Review by indikasampath2000 — APPROVED** [review]

**Pull request merged**